### PR TITLE
Swipl API updates 9.1.19

### PR DESCRIPTION
**Pull request by morlic**

### CODE REVIEW

**Comment by morlic** [comment]
Not sure how to do this for backward compatibility, maybe check if counts fit in `int32` as before in `Context::new_term_refs`?

**Comment by morlic** [comment]
duplicate/similar to #54

File: swipl/src/dict.rs [hunk 1]
```diff
@@ -206,8 +206,7 @@ unsafe impl<'a> TermPutable for DictBuilder<'a> {
 
         let tag_term = context.new_term_ref();
         let len = self.entries.len();
-        // TODO assert len is not too big
-        let value_terms = unsafe { fli::PL_new_term_refs(len as i32) };
+        let value_terms = unsafe { fli::PL_new_term_refs(len) };
```
Comment (matko) [inline swipl/src/dict.rs:209]: Probably we should have a `.try_into().unwrap()` here as well, right? This seems to be the way to make sure things work regardless of the generated signature of `PL_new_term_refs`.

**Comment by matko** [comment]
This now appears to build both 9.0 and 9.2 on my machine and gets through the CI.

Thank you for your PR and apologies for taking so long to get to this.

File: swipl/src/dict.rs (hunk 1 above)
Comment (morlic) [reply swipl/src/dict.rs]: yep, that would make it work towards legacy swipl that expects i32

**Pull request merged**